src/client/shadows/dynamicshadows.cpp

```diff
@@ -68,11 +68,15 @@ void DirectionalLight::createSplitMatrices(const Camera *cam)
 	// we must compute the viewmat with the position - the camera offset
 	// but the future_frustum position must be the actual world position
 	v3f eye = center_scene - eye_displacement;
+	v3f up = v3f(0.0f, 1.0f, 0.0f);
+	// eye_displacement and up shall not be collinear
+	if (core::equals(eye_displacement.crossProduct(up).getLengthSQ(), 0.f))
```

Comment:
comment should maybe explain that if they are, that is, the sun is shining ("looking") straight down at high noon, it doesn't really matter which vector we choose so long as it is orthogonal to the sun's look direction (or does it? and if so, why do we have to choose this vector?)